### ISSUE TITLE
EarlyStopping with learning rate consideration 

### ISSUE THREAD

**Issue by chrstn-hntschl** [chrstn-hntschl/deepvcd#8]
EarlyStopping in Keras only supports epoch based patience (if metric does not improve after n epochs by a given value, learning is stopped). This may interfere with ReduceLROnPlateau: either we might stop learning before reduction of learning rate has stopped fluctuating or we have a rather high patience for early stopping. An alternative would be to have early stopping only active when a given learning rate has been reached.

**Comment by chrstn-hntschl** [comment]
Can be handled to some extend using improved `ReduceLROnPlateau` and `EarlyStopping` parameters.

**Comment by chrstn-hntschl** [comment]
Can be handled to some extend using improved `ReduceLROnPlateau` and `EarlyStopping` parameters.
